keras_cv/layers/object_detection/nms_prediction_decoder.py

```diff
@@ -89,28 +89,33 @@ def _decode_box_predictions(self, anchor_boxes, box_predictions):
         )
         return boxes
 
-    def call(self, images, predictions):
+    def call(self, images, predictions, anchor_boxes=None):
         """Accepts images and raw predictions, and returns bounding box predictions.
 
         Args:
             images: Tensor of shape [batch, height, width, channels].
             predictions: Dense Tensor of shape [batch, anchor_boxes, 6] in the
                 `bounding_box_format` specified in the constructor.
+            anchor_boxes: (optional) dense tensor of shape [batch, anchor_boxes, 4].
+                When specified, use these custom anchor boxes instead of using
+                the anchor generator.
         """
         if isinstance(images, tf.RaggedTensor):
             raise ValueError(
                 "DecodePredictions() does not support tf.RaggedTensor inputs. "
                 f"Received images={images}."
             )
 
-        anchor_boxes = self.anchor_generator(images[0])
-        anchor_boxes = tf.concat(list(anchor_boxes.values()), axis=0)
-        anchor_boxes = bounding_box.convert_format(
-            anchor_boxes,
-            source=self.anchor_generator.bounding_box_format,
-            target="xywh",
-            images=images[0],
-        )
+        if anchor_boxes is None:
+            anchor_boxes = self.anchor_generator(images[0])
+            anchor_boxes = tf.concat(list(anchor_boxes.values()), axis=0)
+            anchor_boxes = bounding_box.convert_format(
+                anchor_boxes,
+                source=self.anchor_generator.bounding_box_format,
+                target="xywh",
+                images=images[0],
+            )
+            anchor_boxes = anchor_boxes[None, ...]
```

Comment:
This is functionally equivalent to
```
anchor_boxes = tf.expand_dims(anchor_boxes, axis=0)
```
I just copied it as-is because this is how it was before. It's required because box_predictions has a batch axis, so anchor boxes needs one too

Reply:
hmmm...can we use anchor_boxes[tf.newaxis, ...] instead then?

Reply:
yeah that seems much more readable -- done